lib/fastlane/plugin/mango/helper/docker_commander.rb

```diff
@@ -17,22 +17,29 @@ def pull_image(docker_image_name:)
             Actions.sh("docker pull #{docker_image_name}")
           rescue StandardError
             retry
-          end  
+          end
         end
       end
 
-      def start_container(emulator_args:, docker_image:)
+      def start_container(emulator_args:, docker_image:, core_amount:)
         docker_name = if container_name
                         "--name #{container_name}"
                       else
-                        ''
+                        ""
+                      end
+        # if core_amount value is defined then limit the container while starting
+        core_amount = if !core_amount.nil? && core_amount > 0
```

Comment:
```suggestion
        core_amount = if core_amount > 0
```

Reply:
why do we need to check nil? it must not be nil ever, or you experienced another cases?

Reply:
it is because for the test cases. When we provide the `core_amount` from `Fastline` file it is automatically converted to `0`. But it is not the same case for testing when you provide `nil`, an exception is thrown `nil>0` you cannot compare etc. If we sent this value from tests by converting to integer. It will be again fine I guess.

Reply:
yes updated the if condition, `if core_amount`  already checks null which is cool!